geotrek/common/views.py

```diff
@@ -316,6 +325,87 @@ def get(request, *args, **kwargs):
         return JsonResponse(response)
 
 
+class HDViewPointAPIViewSet(APIViewSet):
+    model = HDViewPoint
+    serializer_class = HDViewPointAPISerializer
+    geojson_serializer_class = HDViewPointAPIGeoJSONSerializer
+
+    def get_queryset(self):
+        return HDViewPoint.objects.annotate(api_geom=Transform("geom", settings.API_SRID))
+
+
+class HDViewPointDetail(CompletenessMixin, mapentity_views.MapEntityDetail):
+    model = HDViewPoint
+    queryset = HDViewPoint.objects.all().select_related('content_type', 'license')
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
```

Comment:
best practice is not to use method_decorator on dispatch here, but LoginRequiredMixin on view